src/translate.py

```diff
@@ -733,11 +733,23 @@ def f64(left: Expression, op: str, right: Expression) -> "BinaryOp":
             floating=True,
         )
 
-    def is_boolean(self) -> bool:
+    def is_comparison(self) -> bool:
         return self.op in ["==", "!=", ">", "<", ">=", "<="]
 
     def negated(self) -> "Condition":
-        if not self.is_boolean() or (
+        if (
+            self.op in ["&&", "||"]
+            and isinstance(self.left, Condition)
+            and isinstance(self.right, Condition)
+        ):
+            # DeMorgan's Laws
+            return BinaryOp(
+                left=self.left.negated(),
+                op={"&&": "||", "||": "&&"}[self.op],
+                right=self.right.negated(),
+                type=Type.bool(),
+            )
```

Comment:
@simonlindholm or @matt-kempster - do you know if this was *intentionally* unimplemented? Were there cases when you could tell IDO didn't simplify the boolean expression? 

From my testing, in conditionals, it looks like IDO always followed the simplification in order to make the control flow graph work. 

Having this done after constructing the conditional is really convenient: there's less state to pass around as the conditional is constructed.